.github/workflows/build_docs.yml

```diff
@@ -0,0 +1,42 @@
+# Please keep in sync with /environment.yml
+name: Build Docs
+
+on: workflow_dispatch
+
+jobs:
+  build-docs:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - name: set git user
+      run: |
+        git config --global user.email "metomi@metoffice.gov.uk"
+        git config --global user.name "SciFab Developers"
+    - uses: actions/checkout@v3
+    - uses: actions/setup-python@v4
+      with:
+        python-version: '3.x'
+    - name: install sphinx
+      run: pip install sphinx sphinx-rtd-theme
+    - name: install fab
+      run: pip install .
+    - name: build docs
+      run: |
+        cd docs
+        rm -rf build
+        sphinx-apidoc -f -o source/apidoc ../source/fab
+        make html
+    - name: move built docs to docs root
+      run: |
+        mv docs/build/html/* docs/
+    - name: git add built docs
+      run: |
+        git add docs/*
+    - name: commit
+      run: |
+        git commit -m "docs build"
```

Comment:
I'm always wary of committing generated files to a source repository. Is there a reason not to build and commit in the `gh_pages` branch?

Reply:
The risk seems the same. Perhaps I'm missing something?
 - Checkout the given branch --> build --> force push to gh_pages
 - Checkout the given branch --> force replace gh_pages --> build --> push

Reply:
The way I did it with Stylist was to checkout both the project and the gh_pages. Then I could copy across. The whole gh_pages thing is rather inelligant whichever way it's done but it's what we have.